PROTO_tests/tests/join_test.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+"""
+Encapsulates a variety of arkouda join_on_eq_with_dt test cases.
+"""
+
+
+class TestJoin:
+    @classmethod
+    def setup_class(cls):
+        cls.size = 1000
+        cls.a1 = ak.ones(cls.size, dtype=np.int64)
+        cls.a2 = ak.arange(0, cls.size, 1)
+        cls.t1 = cls.a1
+        cls.t2 = cls.a1 * 10
+        cls.dt = 10
+        ak.verbose = False
+
+    @pytest.mark.parametrize("dt_type", ["true_dt", "abs_dt", "pos_dt"])
+    def test_join_on_eq_by_dt(self, dt_type):
+        x, y = ak.join_on_eq_with_dt(self.a2, self.a1, self.t1, self.t2, self.dt, dt_type)
+        assert self.size // pytest.nl, x.size
+        assert self.size // pytest.nl, y.size
```

Comment:
so this actually isn't testing anything. It needs to be `assert self.size // pytest.nl == x.size`. Otherwise it's just checking the truthyness of `self.size // pytest.nl` which will be true as long as it's non-zero

To show this:
```python
# this outputs nothing because 17 is truthy
>>> assert 17, 9

# raises an assertion error because 17 == 9 is false
>>> assert 17 == 9
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 assert 17 == 9

AssertionError:

# raises an assertion error because 0 is falsey
>>> assert 0, 17
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[7], line 1
----> 1 assert 0, 17

AssertionError: 17
```

Reply:
Updated